### PR TITLE
make stormpath filter conditional on missing bean,  expos…

### LINKED ISSUE #964: Make Stormpath Filter conditional on missing bean

### PR DESCRIPTION
**Pull request by iancorcoran**
Fixes #964 

…ed some filter dependencies as beans

### CODE REVIEW

File: extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebMvcConfiguration.java [hunk 1]
```diff
@@ -836,6 +837,7 @@ public RefreshTokenResultFactory stormpathRefreshTokenResultFactory() {
         return new DefaultRefreshTokenResultFactory(application);
     }
 
+    @Bean
```
Comment (mrioan) [inline extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebMvcConfiguration.java]: @iancorcoran this is an abstract class. We do not define beans here. All our configurations in the web layer follow the same pattern. We have an abstract class where we implement most of the functionality and 2 concrete classes: one for Spring (in this case `StormpathWebMvcConfiguration.java`) and one for Spring Boot (StormpathWebMvcAutoConfiguration.java). The Beans are defined in these 2 clases, not in the abstract one.

File: extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebMvcAutoConfiguration.java [hunk 2]
```diff
@@ -654,6 +654,7 @@ public FilterChainManager stormpathFilterChainManager() {
     }
 
     @Bean
+    @ConditionalOnMissingBean
```
Comment (mrioan) [inline extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebMvcAutoConfiguration.java]: Please make it `@ConditionalOnMissingbean(name = "stormpathServletContextListener")`. The lack of this explicit naming causes some issues when developers use a different version of Spring.
Additionally, if it is not much trouble, could you please add the name to some of the pre-existent beans that are missing the name please? For example, `stormpathAccessTokenResultFactory`, `stormpathFieldValueResolver`, etc do not have the name explicitly defined (there are quite a few more). Thanks in advance!!

**Comment by mrioan** [comment]
@iancorcoran back to you. Thanks

**Comment by iancorcoran** [comment]
@mrioan I moved bean declarations out of AbstactStormpathWebMvcConfiguration to the subclasses, and also added names to all the other ConditionalOnMissingBean annotations. Let me know if you have any further comments. Thanks!

**Comment by mrioan** [comment]
Looks good do me. Feel free to merge when Travis passes. Thanks!. Back to you @iancorcoran

**Pull request merged**